web/docs/auth/migrate-from-11.md

```diff
@@ -0,0 +1,318 @@
+  userId      Int
+  createdAt   DateTime  @default(now())
+  @@unique([provider, providerId, userId])
+psl=}
+```
+
+### New Auth Models
+
+From 0.12.X onwards, authentication is based on the auth models which are automatically set up by Wasp. You don't need to take care of the auth fields anymore.
+
+The `User` model is now just a business logic model and you use it for storing the data that is relevant for your app.
+
+```wasp title="main.wasp"
+app myApp {
+  wasp: {
+    version: "^0.12.0"
+  },
+  title: "My App",
+  auth: {
+    userEntity: User,
+    methods: {
+      gitHub: {}
+    },
+    onAuthFailedRedirectTo: "/login"
+  },
+}
+
+entity User {=psl
+  id Int @id @default(autoincrement())
+psl=}
+```
+
+You can read more about the new auth system in the [Auth Entities](./entities) section.
+
+## How to Migrate?
+
+Migrating your existing app to the new auth system is a two-step process:
+1. Migrate to the new auth system
+1. Cleanup the old auth system
+
+:::caution Migrating a deployed app
+
+To migrate a deployed app, you need to go through the migration steps while deploying the app in between the steps.
+
+Read more about migrating a deployed app in the [Migrating a Deployed App](#migrating-a-deployed-app) section.
+:::
+
+### 1. Migrate to the New Auth System
+
+If you have an existing app with authentication set up, you can follow these steps to migrate to the new auth system:
+
+1. Upgrade Wasp to the latest version
+1. Bump the version to `^0.12.0` in `main.wasp`
+1. Create the new auth tables in the database by running:
+
+  ```bash
+  wasp db migrate-dev
+  ```
+
+1. Create a data migration script in `src/server/migrateToNewAuth.ts`
+
+    - Below you can find [examples of migration scripts](#example-migration-scripts) for each of the auth methods
+
+1. Add the migration script to the `db.seeds` config:
+  ```wasp title="main.wasp"
+  app myApp {
+    wasp: {
+      version: "^0.12.0"
+    },
+    // ...
+    db: {
+      seeds: [
+        import { migrateEmailAuth } from "@server/migrateToNewAuth.js",
+        import { migrateGoogleAuth } from "@server/migrateToNewAuth.js",
+      ]
+    },
+  }
+  ```
+1. Run the migration script by running:
+  ```bash
+  wasp db seed
+  ```
+1. Verify still works as expected
+1. Migrate to using the new auth helper functions for getting the `email` or `username` of the currently logged in user. Read more about the helpers in the [Auth Entities](./entities#accessing-the-auth-fields) section.
+
+### 2. Cleanup the Old Auth System
+
+Now we need to clean up the old auth system:
+
+1. Delete auth-related fields from `User` entity
+1. Remove the `externalAuthEntity` from the `auth` config and the `SocialLogin` entity (if you used it)
+1. Run `wasp db migrate-dev` again to remove the redundant fields from the database
+1. You can now delete the migration script and the `db.seeds` config
+
+
+## Migrating a Deployed App
```

Comment:
I wonder if instead of having this separate section, we should instead add these steps at their right places in the whole flow, and we can say they have to do these if their app is deployed somewhere, if not, they skip it. That sounds more natural to me and easier to follow.